### PR TITLE
index_put_ for complex tensors on CUDA

### PR DESCRIPTION
**Pull request by anjali411**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#51148 index_put_ for complex tensors on CUDA**

Differential Revision: [D26102025](https://our.internmc.facebook.com/intern/diff/D26102025)

### CODE REVIEW

File: aten/src/ATen/native/cuda/Indexing.cu [hunk 1]
```diff
@@ -230,7 +230,7 @@ void index_put_accum_kernel(Tensor & self, const c10::List<c10::optional<Tensor>
            std::min(std::max<int>(1,nElemBefore), at::cuda::getCurrentDeviceProperties()->maxGridSize[2]));
       dim3 block(C10_WARP_SIZE, indices_per_block);
 
-      AT_DISPATCH_ALL_TYPES_AND3(at::ScalarType::Half, at::ScalarType::Bool, at::ScalarType::BFloat16,
+      AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND3(at::ScalarType::Half, at::ScalarType::Bool, at::ScalarType::BFloat16,
```
Comment (albanD) [inline aten/src/ATen/native/cuda/Indexing.cu:233]: Is cdouble included in this case? Should cdouble be included in the tests as well?

Comment (anjali411) [reply aten/src/ATen/native/cuda/Indexing.cu:233]: yeah it is! sure I could include that in the test as well

Comment (anjali411) [reply aten/src/ATen/native/cuda/Indexing.cu:233]: @albanD done!

**Review by albanD — COMMENTED** [review]
LGTM thanks for the update!

**Review by albanD — APPROVED** [review]

**Pull request closed without merging**